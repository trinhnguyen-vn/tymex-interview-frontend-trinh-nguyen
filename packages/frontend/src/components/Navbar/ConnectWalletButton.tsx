'use client';
import React, { FC } from 'react';
import {
    Button,
} from '@mui/material';

const ConnectWalletButton: FC<{ handleConnectWallet: () => void }> = ({ handleConnectWallet }) => {
    return <Button
        variant="contained"
        onClick={handleConnectWallet}
    >
        Connect Wallet
    </Button>
}

export default ConnectWalletButton;
