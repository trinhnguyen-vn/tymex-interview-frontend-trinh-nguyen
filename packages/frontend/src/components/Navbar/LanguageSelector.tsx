'use client';
import React, { FC, useState } from 'react';
import {
    Box,
    IconButton,
    Menu,
    MenuItem,
} from '@mui/material';
import LanguageIcon from '@mui/icons-material/Language';
import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
import { LANGUAGES, TypeLanguageValue } from '@/constants/languages';
const LanguageSelector: FC<{ handleSelectLanguage: (language: TypeLanguageValue) => void }> = ({ handleSelectLanguage }) => {
    const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
    const handleLanguageMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
        setAnchorEl(event.currentTarget);
    };

    const handleLanguageMenuClose = () => {
        setAnchorEl(null);
    };
    return <>
        <IconButton
            sx={{ color: '#fff' }}
            onClick={handleLanguageMenuOpen}
        >
            <LanguageIcon />
            <ArrowDropDownIcon />
        </IconButton>
        <Menu
            anchorEl={anchorEl}
            open={Boolean(anchorEl)}
            onClose={handleLanguageMenuClose}
        >
            <Box
                sx={{
                    backgroundColor: "background.default",
                    borderRadius: 1,
                }}
            >
                {LANGUAGES.map((language) => (
                    <MenuItem
                        key={language.value}
                        onClick={() => {
                            handleLanguageMenuClose();
                            handleSelectLanguage(language.value)
                        }}
                        sx={{ color: language.value === 'en' ? 'primary.main' : 'text.secondary', fontWeight: language.value === 'en' ? 'bold' : 'normal', }}
                    >
                        {language.label}
                    </MenuItem>

                ))}   </Box>  </Menu>
    </>
}

export default LanguageSelector;
