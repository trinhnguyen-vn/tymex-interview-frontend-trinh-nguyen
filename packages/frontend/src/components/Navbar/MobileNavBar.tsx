'use client';
import React, { FC, useState } from 'react';
import { useRouter } from 'next/navigation';
import {
    IconButton,
    Drawer,
    List,
    ListItem,
    ListItemText,
} from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import { NAV_LINKS } from '@/constants/navbar';
import ConnectWalletButton from './ConnectWalletButton';
import LanguageSelector from './LanguageSelector';
import { TypeLanguageValue } from '@/constants';

const MobileNavbar: FC<{ handleConnectWallet: () => void; handleSelectLanguage: (language: TypeLanguageValue) => void; checkIsActive: (path: string) => boolean }> = ({ handleConnectWallet, handleSelectLanguage, checkIsActive }) => {
    const router = useRouter();
    const [mobileOpen, setMobileOpen] = useState(false);

    const handleDrawerToggle = () => {
        setMobileOpen(!mobileOpen);
    };

    return <>
        <IconButton
            color="inherit"
            edge="end"
            onClick={handleDrawerToggle}
            sx={{ color: '#fff' }}
        >
            <MenuIcon />
        </IconButton>
        <Drawer
            anchor="right"
            open={mobileOpen}
            onClose={handleDrawerToggle}
            sx={{
                '& .MuiDrawer-paper': {
                    backgroundColor: '#1A1A2E',
                    color: '#fff',
                },
            }}
        >
            <List>
                {NAV_LINKS.map((link) => (
                    <ListItem
                        key={link.label}
                        component="button"
                        onClick={() => {
                            router.push(link.path);
                            setMobileOpen(false);
                        }}
                        sx={{
                            backgroundColor: checkIsActive(link.path) ? 'primary.main' : 'transparent',
                        }}
                    >
                        <ListItemText primary={link.label} />
                    </ListItem>
                ))}
                <ListItem>
                    <ConnectWalletButton handleConnectWallet={handleConnectWallet} />
                </ListItem>
                <ListItem>
                    <LanguageSelector handleSelectLanguage={handleSelectLanguage} />
                </ListItem>
            </List>
        </Drawer>
    </>

}
export default MobileNavbar;