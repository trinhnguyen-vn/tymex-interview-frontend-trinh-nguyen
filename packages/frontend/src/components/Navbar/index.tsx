'use client';
import React, { useEffect, useState } from 'react';
import { usePathname, useRouter } from 'next/navigation';
import {
    AppBar,
    Toolbar,
    Typography,
    useMediaQuery,
    useTheme,
    Box,
} from '@mui/material';
import { StyledLink } from '@/styles/Link';
import { TypeLanguageValue } from '@/types';
import { NAV_LINKS } from '@/constants/navbar';
import MobileNavbar from './MobileNavBar';
import ConnectWalletButton from '@/components/Wallets/ConnectWalletButton';
import LanguageSelector from './LanguageSelector';
import { MAIN_BACKGROUND_COLOR } from '@/styles';

const Navbar: React.FC = () => {
    const router = useRouter();
    const theme = useTheme();
    const isMobile = useMediaQuery(theme.breakpoints.down('md'));
    const currentActivePath = usePathname();
    const checkIsActive = (path: string) => currentActivePath === path;
    const [scrolled, setScrolled] = useState(false);

    useEffect(() => {
        const handleScroll = () => {
            setScrolled(window.scrollY > 20);
        };
        window.addEventListener('scroll', handleScroll);
        return () => window.removeEventListener('scroll', handleScroll);
    }, []);

    const handleConnectWallet = () => {
        // wallet connection logic here
        console.log('Connecting wallet...');
        router.push('/wallet');
    };

    const handleSelectLanguage = (language: TypeLanguageValue) => {
        console.log('Selecting language...', language);
    };

    return (
        <AppBar
            position="fixed"
            sx={{
                background: MAIN_BACKGROUND_COLOR,
                borderBottom: scrolled ? `3px solid ${theme.palette.primary.main}` : 'none',
            }}
        >
            <Toolbar sx={{ justifyContent: 'space-between', height: '80px', }}>
                {/* Logo */}
                <StyledLink href="/">
                    <Typography variant="h6" sx={{ color: '#fff', fontWeight: 'bold' }}>
                        MarketPlace
                    </Typography>
                </StyledLink>

                {isMobile ? (
                    // Mobile Navigation
                    <MobileNavbar checkIsActive={checkIsActive} handleConnectWallet={handleConnectWallet} handleSelectLanguage={handleSelectLanguage} />
                ) : (
                    <>
                        {/* Desktop Navigation  */}
                        <Box sx={{ display: 'flex', gap: 5 }} data-testid="desktop-navbar">
                            {NAV_LINKS.map((link) => (
                                <StyledLink href={link.path} passHref key={link.label} sx={{
                                    color: checkIsActive(link.path) ? 'primary.main' : 'text.primary', textTransform: "uppercase"
                                }}>
                                    {link.label}
                                </StyledLink>
                            ))}
                        </Box>

                        <div>
                            <ConnectWalletButton handleConnectWallet={handleConnectWallet} />
                            <LanguageSelector handleSelectLanguage={handleSelectLanguage} />

                        </div>
                    </>
                )}
            </Toolbar>
        </AppBar>
    );
};

export default Navbar;