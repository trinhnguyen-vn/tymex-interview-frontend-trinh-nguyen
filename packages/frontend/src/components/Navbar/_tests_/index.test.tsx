import { render, screen } from '@testing-library/react';
import Navbar from '@/components/Navbar';

describe('Navbar', () => {
    let useMediaQueryMock: jest.SpyInstance;
    afterEach(() => {
        if (useMediaQueryMock) useMediaQueryMock.mockRestore();
    });

    describe('Navbar Desktop', () => {
        it('renders Navbar Desktop correctly', () => {
            useMediaQueryMock = jest.spyOn(require('@mui/material/useMediaQuery'), 'default').mockReturnValue(false);

            const { asFragment } = render(<Navbar />);

            expect(screen.getByTestId('desktop-navbar')).toBeInTheDocument();
            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe('Mobile responsive', () => {
        it('renders MobileNavbar Mobile correctly', () => {
            useMediaQueryMock = jest.spyOn(require('@mui/material/useMediaQuery'), 'default').mockReturnValue(true);

            const { asFragment } = render(<Navbar />);

            expect(screen.getByTestId('mobile-navbar')).toBeInTheDocument();
            expect(asFragment()).toMatchSnapshot();
        });
    });
});
