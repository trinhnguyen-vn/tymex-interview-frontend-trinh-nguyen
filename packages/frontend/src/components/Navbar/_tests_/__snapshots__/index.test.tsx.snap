// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Mobile responsive renders MobileNavbar Mobile correctly 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1xhqa78-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-14p6cgn-MuiToolbar-root"
    >
      <a
        class="css-g70xym"
        href="/"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2nr45v-MuiTypography-root"
        >
          MarketPlace
        </h6>
      </a>
      <div
        data-testid="mobile-navbar"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1szr4b2-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
            />
          </svg>
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Navbar Navbar Desktop renders Navbar Desktop correctly 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1xhqa78-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-14p6cgn-MuiToolbar-root"
    >
      <a
        class="css-g70xym"
        href="/"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2nr45v-MuiTypography-root"
        >
          MarketPlace
        </h6>
      </a>
      <div
        class="MuiBox-root css-16wajk8"
        data-testid="desktop-navbar"
      >
        <a
          class="css-obdbpf"
          href="/"
        >
          Home
        </a>
        <a
          class="css-60p31l"
          href="/about"
        >
          About us
        </a>
        <a
          class="css-60p31l"
          href="/teams"
        >
          Our teams
        </a>
        <a
          class="css-60p31l"
          href="/marketplace"
        >
          Marketplace
        </a>
        <a
          class="css-60p31l"
          href="/roadmap"
        >
          Roadmap
        </a>
        <a
          class="css-60p31l"
          href="/whitepaper"
        >
          Whitepaper
        </a>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Connect Wallet
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1hxj9v2-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="LanguageIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56M12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96M4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2s.06 1.34.14 2zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56m2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8M12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96M14.34 14H9.66c-.09-.66-.16-1.32-.16-2s.07-1.35.16-2h4.68c.09.65.16 1.32.16 2s-.07 1.34-.16 2m.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56M16.36 14c.08-.66.14-1.32.14-2s-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m7 10 5 5 5-5z"
            />
          </svg>
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
